### PR TITLE
Add --log-level option.

### PR DESCRIPTION
**Pull request by aheinz-fe**
Only wrinkle is that I wrote the test with pytest, which is what HIP uses for all of our python projects. It's fast and super easy to integrate with TeamCity, including code coverage.

### CODE REVIEW

**Comment by aheinz-fe** [comment]
Oh good, it looks like py.test is what Travis is running anyway.

File: synapse/tests/tools/test_dmon.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+
+from synapse.tools.dmon import getArgParser
+import pytest
```
Comment (invisig0th) [inline synapse/tests/tools/test_dmon.py]: While travis-ci ( and maybe even AppVeyor ) have pytest present in their test environments, it's also expected that a clean checkout should run correctly on other platforms.  Minimizing dependancies is a huge goal of synapse so I think in this case adding an entire 3rd party dependency for this one test is probably not compelling enough :)  For consistency sake, would you mind implementing a quick unit test that lives in the same module but extends SynTest ( from synapse.tests.common ) using only the "unittest" builtins and/or synapse itself?  :)

Comment (invisig0th) [inline synapse/tests/tools/test_dmon.py]: Other than that, this looks great.  With those changes i'll mainline this PR today :)

File: .gitignore [hunk 2]
```diff
@@ -55,3 +55,6 @@ docs/_build/
 
 # PyBuilder
 target/
+
+# pyvenv
+venv
```
Comment (invisig0th) [inline .gitignore]: what's this guy?

Comment (aheinz-fe) [inline .gitignore]: Oh, I do all my python development in virtual environments. Basically
```
pyvenv --clear venv
source venv/bin/activate
pip install --upgrade pip
pip install --upgrade setuptools
python setup.py develop
```
gets me an encapsulated development environment, then
```
python setup.py test
```
installs test_requires (not currently defined in setup.py) and runs tests.

File: synapse/tests/tools/test_dmon.py (hunk 1 above)
Comment (aheinz-fe) [inline synapse/tests/tools/test_dmon.py]: Will do.

**Pull request merged**